hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/applicationsmanager/TestAMRMRPCNodeUpdates.java

```diff
@@ -98,6 +111,47 @@ public AllocateResponse run() throws Exception {
     });
   }
 
+  @Test
+  public void testAMRMRecommissioningNodes() throws Exception {
+    MockNM nm1 = rm.registerNode("127.0.0.1:1234", 10000);
+    MockNM nm2 = rm.registerNode("127.0.0.2:1234", 10000);
+    rm.drainEvents();
+
+    RMApp app1 = rm.submitApp(2000);
+
+    // Trigger the scheduling so the AM gets 'launched' on nm1
+    nm1.nodeHeartbeat(true);
+
+    RMAppAttempt attempt1 = app1.getCurrentAppAttempt();
+    MockAM am1 = rm.sendAMLaunched(attempt1.getAppAttemptId());
+
+    // register AM returns no unusable node
+    am1.registerAppAttempt();
+
+    // DECOMMISSION nm2
+    syncNodeGracefulDecommission(nm2);
+
+    AllocateRequest allocateRequest1 =
+            AllocateRequest.newInstance(0, 0F, null, null, null);
+    AllocateResponse response1 =
+            allocate(attempt1.getAppAttemptId(), allocateRequest1);
+    List<NodeReport> updatedNodes = response1.getUpdatedNodes();
+    Assert.assertEquals(0, updatedNodes.size());
```

Comment:
I noticed the size of `updatedNodes` is 1 in #2564. It is because YARN-6483 is not in branch-2.10.
Now I'm thinking we need to backport YARN-6483 before YARN-10538. @ashutoshcipher what do you think?

Reply:
Yeah @aajisaka -  this makes sense. Let's try back porting [YARN-6483](https://issues.apache.org/jira/browse/YARN-6483) and see if there aren't much dependencies to be backported along with it and we can successfully do that.

Reply:
@aajisaka - Updated the test case with `updatedNodes with 1 `and its passing - checked in my local after [YARN-6483](https://issues.apache.org/jira/browse/YARN-6483) is backport to branch-2.10